src/crynamo/marshaller.cr

```diff
@@ -48,32 +48,92 @@ module Crynamo
       Hash.zip(keys, dynamodb_values)
     end
 
+    def first_key(value : JSON::Any)
```

Comment:
not the happiest with this solution. I tried separating the logic to objects but decided it was better to start with this.